src/ACEtk/ContinuousEnergyTable/src/setSZA.hpp

```diff
@@ -0,0 +1,50 @@
+void setSZA() {
+
+  // check for S, Z, A in NXS(9), NXS(10) and NXS(11)
+  // using MCNP rules
+  if ( ( this->data().NXS(9) == 0 ) &&
+       ( this->data().NXS(10) == 0 ) &&
+       ( this->data().NXS(11) == 0 ) ) {
+
+    std::string zaid = this->ZAID();
+    unsigned int za = std::stoi( zaid.substr( 0, zaid.size() - 4 ) );
+    unsigned int a = za % 1000;
+    unsigned int z = ( za - a ) / 1000;
+    unsigned int s = 0;
+    if ( z > 999 ) {
+
+      unsigned int sz = z;
+      z = z % 1000;
+      s = ( sz - z ) / 1000;
+    }
+
+    if ( a != 999 ) {
```

Comment:
Also, sorry I misspoke when I was explaining what was in our internals. A Z of 999 is "nonphysical". We have test data of ZAID = 999902.01c and stuff.

Reply:
OK, so in this case, does it make sense to try and derive a proper A value? I can just change this line to if (z != 999) and be done with it.

Reply:
Yes this is good.